### ISSUE TITLE
viperHTML framework

### ISSUE THREAD

**Issue by ansarizafar** [WebReflection/viperHTML#3]
Is there any plan to develop viperHtml as a complete framework with routing, state management and model based validation like VueJs.

**Comment by WebReflection** [comment]
`viperHTML` is a rendering engine. It's a bit like asking if [mustache](https://mustache.github.io/) will ever be a complete framework.

The TL;DR answer is then that if that would happen, most likely it won't happen here.

I want to keep concerns separated and do one thing only here (or `hyperHTML` repo).

However, there are interesting standalone routing libraries such [router](https://github.com/tildeio/router.js/) and [page](https://github.com/visionmedia/page.js), which purpose is to handle the routing part.

Those would play well with `hyperHTML` and I've created a [twitter poll](https://twitter.com/WebReflection/status/848146975639953409) to see what developers prefer.

For state management, there are also libraries created at hoc for such purpose, including [redux](https://www.sitepoint.com/redux-without-react-state-management-vanilla-javascript/)
